go/vt/vtgate/vtgateservice/interface.go

```diff
@@ -32,10 +31,9 @@ import (
 // VTGateService is the interface implemented by the VTGate service,
 // that RPC server implementations will call.
 type VTGateService interface {
-	// V3 API
-	Execute(ctx context.Context, session *vtgatepb.Session, sql string, bindVariables map[string]*querypb.BindVariable) (*vtgatepb.Session, *sqltypes.Result, error)
+	Execute(ctx context.Context, mysqlCtx MySQLConnection, session *vtgatepb.Session, sql string, bindVariables map[string]*querypb.BindVariable) (*vtgatepb.Session, *sqltypes.Result, error)
 	ExecuteBatch(ctx context.Context, session *vtgatepb.Session, sqlList []string, bindVariablesList []map[string]*querypb.BindVariable) (*vtgatepb.Session, []sqltypes.QueryResponse, error)
```

Comment:
shouldn't executeBatch also take a MySQLConnection?

Reply:
plugin_mysql_server does not call `ExecuteBatch`. Therefore it was not considered for changes.